content/courses/advanced-ada/parts/data_types/shared_variable_control.rst

```diff
@@ -469,20 +473,806 @@ these array declarations are equivalent:
+
+    pragma Ada_2022;
+
+    with Registers;   use Registers;
+
+    procedure Show_Register is
+       WR : Window_Register :=
+              (Horizontal_Cnt => 800,
+               Vertical_Cnt   => 600,
+               Refresh_Needed => True,
+               others         => <>);
+    begin
+       WR := (Horizontal_Cnt =>
+                WR.Horizontal_Cnt * 2,
+              Vertical_Cnt   =>
+                Wr.Vertical_Cnt   * 2,
+              others         => <>);
+
+       Show (WR);
+    end Show_Register;
+
+In this example, we have an initialization using the same aggregate as in the
+previous code example. We also have an assignment, in which we read the value
+of :ada:`WR` and use it in the calculation.
+
+
+.. _Adv_Ada_Shared_Variable_Control_Non_Atomic_Full_Access_Delta_Aggregates:
+
+Delta aggregates
+^^^^^^^^^^^^^^^^
+
+If we want to just change two components, but leave the information of other
+components untouched, we can use a
+:ref:`delta aggregate <Adv_Ada_Delta_Aggregates>`.
+(Note that we haven't discussed the topic of delta aggregates yet: we'll do
+that :ref:`later on in this course <Adv_Ada_Delta_Aggregates>`. However, in
+simple terms, we can use them to modify specific components of a record without
+changing the remaining components of the record.)
+
+For example, we might want to update just the vertical count and indicate that
+update via the :ada:`Refresh_Needed` flag, but keep the same horizontal count:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Data_Types.Type_Representation.Shared_Variable_Control.Nonatomic_Full_Access_Register
+
+    pragma Ada_2022;
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    with Registers;   use Registers;
+
+    procedure Show_Registers is
+       WR : Window_Register :=
+              (Horizontal_Cnt => 800,
+               Vertical_Cnt   => 600,
+               others         => <>);
+    begin
+       --  Delta assignment
+       WR := (WR with delta
+                   Vertical_Cnt   => 800,
+                   Refresh_Needed => True);
+
+       Show (WR);
+    end Show_Registers;
+
+A delta assignment using an aggregate such as :ada:`(WR with delta ...)`
+includes reading the value of complete 32-bit :ada:`WR` object from memory,
+changing the components specified after :ada:`with delta`, and writing the
+complete 32-bit :ada:`WR` object back to memory. The reason is that we need to
+retrieve the information that is supposed to remain intact |mdash| the
+:ada:`Horizontal_Cnt` and the reserved components |mdash| in order to write
+them back as a *full-access* operation.
+
+
+Atomic full-access
+~~~~~~~~~~~~~~~~~~
+
+As we already know,
+:ref:`atomic objects <Adv_Ada_Shared_Variable_Control_Atomic>` only accept
+atomic reads and updates, which |mdash| as a whole |mdash| are indivisible,
+i.e. they must be done in a single instruction, so that no other instruction
+could execute on that same object before the read or update completes.
+
```

Comment:
Maybe a reminder that atomic implies volatile should go here (instead of above)?

Reply:
Yes, it's even better to just repeat it here as well.